server/src/test/java/org/elasticsearch/search/fetch/subphase/FieldFetcherTests.java

```diff
@@ -394,21 +397,288 @@ public void testTextSubFields() throws IOException {
+     */
+    public void testMappedFieldNotOverwritten() throws IOException {
+        XContentBuilder mapping = XContentFactory.jsonBuilder().startObject()
+            .startObject("properties")
+                .startObject("f1")
+                    .field("type", "integer")
+                    .field("ignore_malformed", "true")
+                .endObject()
+            .endObject()
+        .endObject();
+
+        IndexService indexService = createIndex("index", Settings.EMPTY, mapping);
+        MapperService mapperService = indexService.mapperService();
 
+        XContentBuilder source = XContentFactory.jsonBuilder().startObject()
+            .field("f1", "malformed")
+            .endObject();
+
+        // this should not return a field bc. f1 is in the ignored fields
+        Map<String, DocumentField> fields = fetchFields(
+            mapperService,
+            source,
+            List.of(new FieldAndFormat("*", null, Optional.of(true))),
+            Set.of("f1")
+        );
+        assertThat(fields.size(), equalTo(0));
+
+        // and this should neither
+        fields = fetchFields(mapperService, source, List.of(new FieldAndFormat("*", null, Optional.of(true))), Set.of("f1"));
+        assertThat(fields.size(), equalTo(0));
+
+        fields = fetchFields(mapperService, source, List.of(new FieldAndFormat("f1", null, Optional.of(true))), Set.of("f1"));
+        assertThat(fields.size(), equalTo(0));
+    }
+
+    public void testUnmappedFieldsWildcard() throws IOException {
+        MapperService mapperService = createMapperService();
+
+        XContentBuilder source = XContentFactory.jsonBuilder().startObject()
+            .startObject("unmapped_object")
+                .field("a", "foo")
+                .field("b", "bar")
+            .endObject()
+            .endObject();
+
+        Map<String, DocumentField> fields = fetchFields(mapperService, source, "unmapped_object", true);
+        assertThat(fields.size(), equalTo(0));
+
+        fields = fetchFields(mapperService, source, "unmap*object", true);
+        assertThat(fields.size(), equalTo(0));
+
+        fields = fetchFields(mapperService, source, "unmapped_object.*", true);
+        assertThat(fields.size(), equalTo(2));
+        assertThat(fields.keySet(), containsInAnyOrder("unmapped_object.a", "unmapped_object.b"));
+
+        assertThat(fields.get("unmapped_object.a").getValue(), equalTo("foo"));
+        assertThat(fields.get("unmapped_object.b").getValue(), equalTo("bar"));
+
+        fields = fetchFields(mapperService, source, "unmapped_object.a", true);
+        assertThat(fields.size(), equalTo(1));
+        assertThat(fields.get("unmapped_object.a").getValue(), equalTo("foo"));
+
+        fields = fetchFields(mapperService, source, "unmapped_object.b", true);
+        assertThat(fields.size(), equalTo(1));
+        assertThat(fields.get("unmapped_object.b").getValue(), equalTo("bar"));
+    }
+
+    private Map<String, DocumentField> fetchFields(
```

Comment:
Small comment: there's getting to be a lot of convenience methods, maybe we could prefer just constructing and passing in `FieldAndFormat`?

Reply:
Removed some